.github/workflows/haros.yml

```diff
@@ -0,0 +1,58 @@
+name: Haros Analysis
+
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    branches: [ develop ]
+
+jobs:
+  haros:
+    runs-on: ubuntu-20.04
+    container:
+      image: swarmus/embedded
+    steps:
+      - name: Create catkin_ws
+        run: |
+          mkdir -p catkin_ws/src
+          cd catkin_ws
+          . /opt/ros/$ROS_DISTRO/setup.sh
+          catkin_make
+
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          path: 'catkin_ws/src/SwarmUS-ROS'
+
+      - name: Checkout submodules
+        run: git submodule update --init --recursive
+        working-directory: 'catkin_ws/src/SwarmUS-ROS'
+
+      - name: Install Haros
+        run: |
+          . /opt/ros/$ROS_DISTRO/setup.sh
+          apt-get update && apt-get install -y python2 libmagic-dev
+          wget https://bootstrap.pypa.io/get-pip.py
+          python2 get-pip.py
+          pip2 --version
+          pip2 install haros
+
+      - name: Run Haros
+        run: |
+          . /opt/ros/$ROS_DISTRO/setup.sh
+          haros analyse -p src/SwarmUS-ROS/haros_config.yaml -b haros_plugin_radon haros_plugin_ccd haros_plugin_cppcheck haros_plugin_cpplint -n
+          mkdir haros
+          haros export -v ./haros
+        working-directory: 'catkin_ws'
+
+      - name: Prepare Publish Step
+        run: |
+          apt-get update && apt-get install -y rsync
+
+      - name: Publish Result
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          personal_token: ${{ secrets.HAROS_VIZ_TOKEN }}
+          external_repository: SwarmUS/Haros-viz
```

Comment:
I'm just curious, why on an other repository and not just another branch like the other? And what happens if we have multiple repository with the need to analayze ROS packages? Do they all publish to this repo?

Reply:
I didn't look so much into that, but I thought that it was impossible to host two seperate gh pages sites on a single repo. Am I wrong?

Reply:
You are right we can't generate multiple github pages, though could we all put the doc together? as of now we have 3 modules, thus 3 link in your index.html in the doc github pages, can it be possible to add a fourth and it would be haros?

 How is the index.html from the docs ci created? If it's form the github action, probably it's possible to have them all on the same gh page

Reply:
The index.html is generated by one of the python scripts, which is run from github actions. Thing is, the docs and Haros are run from two separate CI pipelines.
I could just create a root index.html and deploy the docs and haros stuff to two separate folders. I wanted to avoid messing with too many html pages, manually though, this is why I did it this way.

Reply:
Can't one CI pipeline generate all the doc? I would definitly prefer all the doc on the github pages but if it's too much hassle maybe just rename the haros doc repo for SwarmUS-ROS-Haros-viz so we know it's associated with it and just in case we have another use for Haros and want to visualise it. 

I'll approve the PR so you can merge once you've decided and done the changes

Reply:
Yeah right now the two CI pipelines do not have the same trigger conditions. I'll just rename the other repo.